azurerm/internal/services/network/private_endpoint_resource.go

```diff
@@ -106,10 +106,22 @@ func resourceArmPrivateEndpoint() *schema.Resource {
 							ForceNew: true,
 						},
 						"private_connection_resource_id": {
```

Comment:
WOuld it make sense to add a new property `private_connection_service_alias` rather then overload this one? - as resource_id is fairly explicit in what values it is expecting.

Reply:
This is how the Azure API works (says one thing, accepts a few things).  Does this need to be more strict than that?

Reply:
https://docs.microsoft.com/en-us/rest/api/virtualnetwork/privateendpoints/createorupdate#request-body

Reply:
hey @dlamotte

Apologies for the delayed reply here.

Chatting about this internally whilst the Azure API exposes a single field for multiple purposes within Terraform we're tending to split these out into separate fields so this is clearer to users and then set the appropriate field in the request/parsing the response back out to two fields here - and we can ensure that users only set a single field by using an `ExactlyOneOf` in the Schema too. As Katie has mentioned the benefit to doing this is primarily advanced validation - as such could we split this out here?

Thanks!

Reply:
Sure.  Sounds good.